dataplane/linux/bpf_ep_mgr_test.go

```diff
@@ -126,27 +217,128 @@ var _ = Describe("BPF Endpoint Manager", func() {
 		},
 	}
 
+	hostEpNorm := proto.HostEndpoint{
+		Name: "uthost-eth0",
+		Tiers: []*proto.TierInfo{
+			&proto.TierInfo{
+				Name:            "mytier",
+				IngressPolicies: []string{"mypolicy"},
+				EgressPolicies:  []string{"mypolicy"},
+			},
+		},
+	}
+
 	It("does not have HEP in initial state", func() {
 		Expect(bpfEpMgr.hostIfaceToEpMap["eth0"]).NotTo(Equal(hostEp))
 	})
 
+	Context("with workload and host-* endpoints", func() {
+		JustBeforeEach(func() {
+			genIfaceUpdate("eth0", ifacemonitor.StateUp, 10)()
+			genWLUpdate("cali12345")()
+			genIfaceUpdate("cali12345", ifacemonitor.StateUp, 15)()
+			genHEPUpdate(allInterfaces, hostEpNorm)()
+		})
+
+		It("does not have host-* policy on the workload interface", func() {
+			Eventually(func() map[string]uint32 {
+				return dp.fds
+			}).Should(HaveLen(4))
+
+			// Check eth0 ingress.
+			eth0I := dp.fds["eth0-I"]
+			Expect(eth0I).NotTo(BeZero())
+			Expect(dp.state).To(HaveKey(eth0I))
+			Expect(dp.state[eth0I].ForHostInterface).To(BeTrue())
+			Expect(dp.state[eth0I].HostNormalTiers).To(HaveLen(1))
+			Expect(dp.state[eth0I].HostNormalTiers[0].Policies).To(HaveLen(1))
+			Expect(dp.state[eth0I].SuppressNormalHostPolicy).To(BeFalse())
+
+			// Check eth0 egress.
+			eth0E := dp.fds["eth0-E"]
+			Expect(eth0E).NotTo(BeZero())
+			Expect(dp.state).To(HaveKey(eth0E))
+			Expect(dp.state[eth0E].ForHostInterface).To(BeTrue())
+			Expect(dp.state[eth0E].HostNormalTiers).To(HaveLen(1))
+			Expect(dp.state[eth0E].HostNormalTiers[0].Policies).To(HaveLen(1))
+			Expect(dp.state[eth0E].SuppressNormalHostPolicy).To(BeFalse())
+
+			// Check workload ingress.
+			caliI := dp.fds["cali12345-I"]
+			Expect(caliI).NotTo(BeZero())
+			Expect(dp.state).To(HaveKey(caliI))
+			Expect(dp.state[caliI].ForHostInterface).To(BeFalse())
+			Expect(dp.state[caliI].SuppressNormalHostPolicy).To(BeTrue())
+
+			// Check workload egress.
+			caliE := dp.fds["cali12345-E"]
+			Expect(caliE).NotTo(BeZero())
+			Expect(dp.state).To(HaveKey(caliE))
+			Expect(dp.state[caliE].ForHostInterface).To(BeFalse())
+			Expect(dp.state[caliE].SuppressNormalHostPolicy).To(BeTrue())
```

Comment:
No, because empty tiers/profiles for normal policy is not the same as being completely skipped or suppressed.  (In fact empty tiers/profiles = Deny.)  So there's no reason for the calling code to zero out the tiers/profiles fields when it sets SuppressNormalHostPolicy = true.